src/components/spatial/Spatial.js

```diff
@@ -217,6 +295,65 @@ export default class Spatial extends AbstractSelectableComponent {
     });
   }
 
+  createTIFFLayer() {
+    const layerType = 'tiff';
+    const source = this.tiff;
+    if (source.height) {
+      const propSettings = {
+        height: source.height * source.tileSize,
+        width: source.width * source.tileSize,
+      };
+      const minZoom = Math.floor(-1 * Math.log2(Math.max(propSettings.height, propSettings.width)));
+      return new BaseTileLayer({
+        id: `${layerType}-tile-layer`,
+        pickable: false,
+        coordinateSystem: COORDINATE_SYSTEM.CARTESIAN,
+        // eslint-disable-next-line  arrow-body-style
+        getTileData: ({ x, y, z }) => {
+          return loadZarr(source.channels, source.tileSize, x, y, -1 * z,
+            Math.ceil(source.width / (2 ** -z)), this.state.gl);
+        },
+        // eslint-disable-next-line  arrow-body-style
+        getTileIndices: (viewport, maxZoomLevel, minZoomLevel) => {
+          return getTileIndices(viewport, maxZoomLevel, minZoomLevel, source.tileSize,
+            source.width * source.tileSize, source.height * source.tileSize);
+        },
+        // eslint-disable-next-line  arrow-body-style
+        tileToBoundingBox: (x, y, z) => {
+          return tileToBoundingBox(x, y, z, propSettings.height,
+            propSettings.width, source.tileSize);
+        },
+        minZoom,
+        maxZoom: 0,
+        visible: true,
+        sliderValues: {
+          redSliderValue: 10000,
+          greenSliderValue: 10000,
+          blueSliderValue: 10000,
+        },
+        renderSubLayers: (props) => {
+          const {
+            bbox: {
+              west, south, east, north,
+            },
+          } = props.tile;
+          const { sliderValues } = props;
+          const xrl = new XRLayer(props, {
+            id: `XR-Layer-${west}-${south}-${east}-${north}`,
```

Comment:
```suggestion
            id: `XR-Layer-w${west}-s${south}-e${east}-n${north}`,
```
Would we ever need to read this?